calico/scripts/prepare-windows-node.ps1

```diff
@@ -0,0 +1,158 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script prepares a Windows node. Its main steps are:
+# - Download the Calico for Windows files needed to install kubelet.
+# - Install and start containerd.
+# - Install and start kubelet.
+#
+# Before running this script:
+# - Prepare a kubeconfig file on the Windows node for kubelet to use.
+#
+# Example usage:
+# .\prepare-windows-node.ps1 -K8sVersion 1.23.4 -DNSNameServers 10.96.0.10
+
+Param(
+    # The version of Kubernetes binaries to install. E.g. 1.23.4"
+    [parameter(Mandatory = $true)] $K8sVersion,
+
+    # The DNS nameserver for kubelet to use. E.g. your cluster's kube-dns cluster IP.
+    [parameter(Mandatory = $true)] $DNSNameServers,
+
+    # The kubeconfig file for kubelet to use.
+    [parameter(Mandatory = $false)] $KubeconfigPath = "c:\k\config",
+
+    # The CNI conf dir to use for containerd.
+    [parameter(Mandatory = $false)] $CNIConfDir = "c:/etc/cni/net.d",
+
+    # The CNI bin dir to use for containerd.
+    [parameter(Mandatory = $false)] $CNIBinDir = "c:/opt/cni/bin",
+
+    # Provide the nodename used by kubelet. If not set, defaults to $(hostname).ToLower()
+    [parameter(Mandatory = $false)] $Nodename = "",
+
+    # The version of containerd to install. E.g. 1.6.2.
+    [parameter(Mandatory = $false)] $ContainerdVersion = "1.6.2",
+
+    # If $true, existing files are overwritten.
+    [parameter(Mandatory = $false)] $Overwrite = $false
+)
+
+$ErrorActionPreference = "Stop"
+$rootDir = "c:\CalicoWindows"
+# Otherwise download takes too long to download
+$ProgressPreference = 'SilentlyContinue'
+$kubeletPath = "c:\k\kubelet.exe"
+
+if ((Get-Service | where Name -EQ 'kubelet') -NE $null) {
+  Write-Host "kubelet service already exists."
+  exit 1
+}
+
+if ((Get-Service | where Name -EQ 'containerd') -NE $null) {
+  Write-Host "containerd service already exists."
+  exit 1
+}
+if (-not (Test-Path $KubeconfigPath)) {
+  Write-Host "The kubeconfig file $KubeconfigPath does not exist. A kubeconfig file must be provided for kubelet."
+  exit 1
+}
+
+if ((Test-Path $kubeletPath) -and (-not $Overwrite)) {
+  Write-Host "kubelet binary already exists at $kubeletPath. Remove it or re-run the script with '-Overwrite true' to overwrite it"
+  exit 1
+}
+
+if ((Test-Path $rootDir) -and (-not $Overwrite)) {
+  Write-Host "$rootDir exists. Remove it or re-run the script with '-Overwrite true' to overwrite those files."
+  Write-Host "WARNING: this will overwrite $rootDir\config.ps1."
+  exit 1
+}
+
+Write-Host "Create root dir $rootDir and supporting files to install kube services..."
+mkdir $rootDir -Force | Out-Null
+mkdir $rootDir\libs\calico -Force | Out-Null
+mkdir $rootDir\kubernetes -Force | Out-Null
+
+$baseUrl = "https://raw.githubusercontent.com/projectcalico/calico/8176616948416e697bb708667dd54ad6fca1884f/node/windows-packaging/"
```

Comment:
This URL is just for ease of testing. I will revert this to point to the release version of the script.